app/src/main/java/android/example/courses/CoursesRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,55 @@
+package android.example.courses
+
+import androidx.recyclerview.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+
+import android.example.courses.CoursesFragment.OnCourseListFragmentListener
+import android.example.courses.dataClasses.Course
+import android.widget.Button
+
+import kotlinx.android.synthetic.main.fragment_item.view.*
+
+class CoursesRecyclerViewAdapter(private val mListenerCourse: OnCourseListFragmentListener?)
+    : RecyclerView.Adapter<CoursesRecyclerViewAdapter.ViewHolder>() {
+
+    private val courses: MutableList<Course>
+    private val mOnClickListener: View.OnClickListener
+
+    init {
+        courses = ArrayList()
+        courses.add(Course(1, "System Design"))
```

Comment:
@radumarin88 The initial plan was to build a backend from scratch, based on Kubernetes, but I realized that it would take me too much time and also I won't be able to learn so many things because I already use Kuberentes at work, so I decided to learn Firebase instead and it turned out to be a very good decision 😄